ui/components/app/hold-to-reveal-button/index.scss

```diff
@@ -0,0 +1,164 @@
+// Variables
+$circle-radius: 14px;
+$circle-diameter: $circle-radius * 2;
+// Circumference ~ (2*PI*R). We reduced the number a little to create a snappier interaction
+$circle-circumference: 82;
+$circle-stroke-width: 2px;
+
+// Keyframes
+@keyframes collapse {
+  from {
+    transform: scale(1);
+  }
+
+  to {
+    transform: scale(0);
+  }
+}
+
+@keyframes expand {
+  from {
+    transform: scale(0);
+  }
+
+  to {
+    transform: scale(1);
+  }
+}
+
+@keyframes fadeIn {
+  from {
+    opacity: 0;
+  }
+
+  to {
+    opacity: 1;
+  }
+}
+
+// Shared styles
+%absolute-fill {
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+}
+
+%icon {
+  height: $circle-diameter;
+  width: $circle-diameter;
+}
+
+%circle-shared {
+  fill: transparent;
+  stroke-width: $circle-stroke-width;
+}
+
+// Class styles
+.button-hold {
```

Comment:
Can we use [BEM](https://en.bem.info/methodology/quick-start/#introduction) here? Remember that all of these styles are global and could therefore clash with other classes we add in the future. So all of these would become

``` sass
.hold-to-reveal-button {
  &__button-hold {
    /* ... */
  }
  
  &__absolute-fill {
    /* ... */
  }

  /* ... etc ... */
}
```

Also is there a reason why we are using `!important`?

Reply:
Fixed in: 2afe9c604ab6d1e79ede7a1d02693f1a34d96e3c
The reason for using the `!important` is 'overriding' some properties of the `Button` component that is being used as the wrapper component in this case.